### DISCUSSION TITLE
Is there any way to create a fixture with a default indirect parameter?

### DISCUSSION THREAD

**Discussion by migue99angel** [pytest-dev/pytest#10798] [Q&A]
I'm thinking about creating a fixture parametrized by the indirect parameter as the documentation specified [here](https://docs.pytest.org/en/7.1.x/example/parametrize.html#indirect-parametrization): 

But  I am wondering if something like this is possible:

```python
@pytest.fixture()
def populate_roles(request) -> list[Role]:
    """
    Generate n roles introduced by an indirect parameter, by default 5 are generated
    """
    created_roles = []
    for test in range(request.param or 5):
       # Code for generating a role

 ```

Actually, I am receiving this error: `AttributeError: 'SubRequest' object has no attribute 'param'`

Thank you! :blush:

**Comment by portyu9** [comment]
Hi @migue99angel 

`request.param` only exists when that fixture has actually been parametrized.

For a fixture that should also work without indirect parametrization, give it an explicit fallback:

```python
@pytest.fixture
def populate_roles(request):
    count = getattr(request, "param", 5)

    created_roles = []
    for _ in range(count):
        # create role
        ...
    return created_roles
```

Then both cases work:

```python
def test_default(populate_roles):
    # uses 5
    ...
```

and:

```python
@pytest.mark.parametrize("populate_roles", [3], indirect=True)
def test_three_roles(populate_roles):
    # uses 3
    ...
```

I would use `getattr(request, "param", 5)` rather than something like `getattr(..., None) or 5`, because the latter also changes a legitimate parameter value of `0` into `5`.

Please mark this answer as accepted if it helped, thank you.
